include/RAJA/pattern/forall.hpp

```diff
@@ -233,16 +254,18 @@ template <typename Res,
           typename SegmentIterPolicy,
           typename SegmentExecPolicy,
           typename LoopBody,
-          typename... SegmentTypes>
+          typename... SegmentTypes,
+          typename ForallParams>
 RAJA_INLINE resources::EventProxy<Res> forall(Res r,
                                          ExecPolicy<SegmentIterPolicy,
                                          SegmentExecPolicy>,
                                          const TypedIndexSet<SegmentTypes...>& iset,
-                                         LoopBody loop_body)
+                                         LoopBody loop_body,
+                                         ForallParams f_params)
```

Comment:
These should be by reference
```suggestion
                                         LoopBody&& loop_body,
                                         ForallParams& f_params)
```